src/models/CustomerUrlData.ts

```diff
@@ -0,0 +1,4 @@
+export interface CustomerUrlData {
+    url: string;
+    auth_headers?: {[header: string]: string};
```

Comment:
Can you explain the purpose of this [header: string] syntax?

Reply:
It forces the type for the dictionary to be string -> string. See `Using an indexed object type annotation` here https://www.carlrippon.com/typescript-dictionary/